plugins/modules/aci_node_block.py

```diff
@@ -0,0 +1,390 @@
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        switch_profile=dict(
+            type="str",
+            aliases=[
+                "leaf_profile_name",
+                "leaf_profile",
+                "switch_profile_name",
+                "spine_switch_profile",
+                "spine_switch_profile_name",
+            ],
+        ),  # Not required for querying all objects
+        access_port_selector=dict(
+            type="str",
+            aliases=[
+                "access_port_selector_name",
+                "port_selector",
+                "port_selector_name",
+            ],
+        ),  # Not required for querying all objects
+        node_block=dict(type="str", aliases=["node_block_name", "name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["node_block_description"]),
+        from_port=dict(type="str", aliases=["from", "from_port_range"]),
+        to_port=dict(type="str", aliases=["to", "to_port_range"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        type_node=dict(type="str", default="leaf", choices=["leaf", "spine"], aliases=["type"]),  # This parameter is not required for querying all objects
```

Comment:
The comment does not make a lot of sense to me if it is not required for query of all objects but the default is always set? Should we enforce a default here, or make it a required_if input for absent/present?